ONSdigital/dp-compose
# create necessary manifests for search stack, and start creating searc…

### What

Search Stack creation, with backend and dependencies only (no frontend)

Depends on the following PRs:
- [Search API](https://github.com/ONSdigital/dp-search-api/pull/202)

### How to review

- Pull the latest develop branches for:
  - `dp-search-api`
  - `dp-search-data-extractor`
  - `dp-search-data-importer`
  - `dp-dataset-api`
  - `zebedee`
  
- in this repo, cd to `/v2/stacks/search`
- run `make start-detached`
- run `watch docker ps -a`

You should see the following docker containers, appearing as healthy:

```
CONTAINER ID   IMAGE                                                  COMMAND                  CREATED          STATUS                    PORTS                                              NAMES
d7856c2fd8b5   confluentinc/cp-kafka:6.0.0                            "/etc/confluent/dock…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:9092->9092/tcp, 19092/tcp                  search-kafka-1-1
801f8f9f0e7b   confluentinc/cp-kafka:6.0.0                            "/etc/confluent/dock…"   13 minutes ago   Up 13 minutes (healthy)   9092/tcp, 19092/tcp, 0.0.0.0:9094->9094/tcp        search-kafka-3-1
5ee52614cf9b   confluentinc/cp-kafka:6.0.0                            "/etc/confluent/dock…"   13 minutes ago   Up 13 minutes (healthy)   9092/tcp, 19092/tcp, 0.0.0.0:9093->9093/tcp        search-kafka-2-1
8ae12b5a4704   quay.io/cloudhut/kowl:master                           "./kowl"                 13 minutes ago   Up 13 minutes             0.0.0.0:9888->8080/tcp                             search-kowl-1
6c93cf8a767f   search-dp-search-data-extractor                        "reflex -d none -c .…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:25800->25800/tcp                           search-dp-search-data-extractor-1
9e5f827e549a   search-dp-dataset-api                                  "reflex -d none -c .…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:22000->22000/tcp                           search-dp-dataset-api-1
fedde9b52086   mongo:3.6                                              "docker-entrypoint.s…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:27017->27017/tcp                           search-mongodb-1
45af29585069   search-zebedee                                         "/bin/sh -c 'java -X…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:8082->8082/tcp, 9200/tcp                   search-zebedee-1
bd77beff9f13   search-dp-search-api                                   "reflex -d none -c .…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:23900->23900/tcp                           search-dp-search-api-1
558e559c8d2c   search-dp-search-data-importer                         "reflex -d none -c .…"   13 minutes ago   Up 13 minutes (healthy)   0.0.0.0:25900->25900/tcp                           search-dp-search-data-importer-1
dc87e18cf43b   docker.elastic.co/elasticsearch/elasticsearch:7.10.0   "/tini -- /usr/local…"   13 minutes ago   Up 13 minutes             0.0.0.0:11200->9200/tcp, 0.0.0.0:11300->9300/tcp   search-sitewideelasticsearch-1
bbb1b9e53040   confluentinc/cp-zookeeper:6.0.0                        "/etc/confluent/dock…"   13 minutes ago   Up 13 minutes             2181/tcp, 2888/tcp, 3888/tcp                       search-zookeeper-1-1
```

### Who can review

Anyone
